cmd/variables.go

```diff
@@ -0,0 +1,288 @@
+			os.Exit(1)
+		}
+		body := strings.Builder{}
+		body.WriteString("Name\tDataService\n")
+		for _, esc := range escResult {
+			body.WriteString(esc.Name + "\t" + esc.DataServiceName + "\n")
+		}
+		if verbose {
+			successMessage := "Successfully retrieved variables from Skuid Site\n"
+			fmt.Println(successMessage + text.Separator() + body.String())
+		} else {
+			fmt.Println(body.String())
+		}
+	},
+}
+
+func getEscs(api *platform.RestApi, mapDsName bool) ([]EnvSpecificConfig, error) {
+	if verbose {
+		fmt.Println(text.VerboseSection("Getting Variables"))
+	}
+
+	escStart := time.Now()
+	api.Connection.APIVersion = "1"
+	result, err := api.Connection.MakeRequest(
+		http.MethodGet,
+		"/ui/variables",
+		nil,
+		"application/json",
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var escs []EnvSpecificConfig
+	if err := json.NewDecoder(*result).Decode(&escs); err != nil { return nil, err }
+
+	if mapDsName {
+		dsmap, err := getDataServices(api)
+		if err != nil { return nil, err }
+		for i, esc := range escs {
+			if esc.DataServiceID != "" {
+				if esc.DataServiceID == fakeDefaultDataServiceId {
+					esc.DataServiceName = "default"
+					escs[i] = esc
+				} else if ds, ok := dsmap[esc.DataServiceID]; ok {
+						esc.DataServiceName = ds.Name
+						escs[i] = esc
+				}
+			}
+		}
+	}
+
+	if verbose {
+		fmt.Println(text.SuccessWithTime("Success getting variable values", escStart))
+	}
+	return escs, nil
+}
+
+// setvarCmd represents the setvariable command
+var setvarCmd = &cobra.Command{
+	Use:   "setvariable",
+	Short: "Set a Skuid environment variable.",
+	Long:  "Set a Skuid envorinment variable. These can be used as credentials for a data source without revealing them.",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		fmt.Println(text.RunCommand("Set Variable"))
+
+		api, err := platform.Login(
+			host,
+			username,
+			password,
+			apiVersion,
+			metadataServiceProxy,
+			dataServiceProxy,
+			verbose,
+		)
+
+		if err != nil {
+			fmt.Println(text.PrettyError("Error logging in to Skuid site", err))
+			os.Exit(1)
+		}
+
+		variableStart := time.Now()
+		err = setEsc(api)
+		if err != nil {
+			fmt.Println(text.PrettyError("Error setting variable in Skuid site", err))
+			os.Exit(1)
+		}
+		successMessage := "Successfully set variable in Skuid Site"
+		if verbose {
+			fmt.Println(text.SuccessWithTime(successMessage, variableStart))
+		} else {
+			fmt.Println(successMessage + ".")
+		}
+	},
+}
+
+func setEsc(api *platform.RestApi) error {
+	if verbose {
+		fmt.Println(text.VerboseSection("Setting Variable"))
+	}
+	if variablename == "" {
+		return errors.New("Variable name is required for this command.")
+	}
+	if variablevalue == "" {
+		fmt.Println("Enter value:")
+		valbytes, err := terminal.ReadPassword(0)
+		if err != nil { return errors.New("Error reading value from prompt.") }
+		variablevalue = string(valbytes)
+	}
+	api.Connection.APIVersion = "1"
+
+	dataServiceId := ""
+	if variabledataservice != "" {
```

Comment:
i think that if `variabledataservice` == "", we should set it to `fakeDefaultDataServiceId`. i didn't expect to have to set it manually to default, and for the majority of skuid sites which don't have Private Data Service enabled, we shouldn't require the use of a data service.